src/Symfony/Component/Translation/Catalogue/DiffOperation.php

```diff
@@ -30,7 +30,7 @@ protected function processDomain($domain)
         );
 
         foreach ($this->source->all($domain) as $id => $message) {
-            if ($this->target->has($id, $domain)) {
+            if (!$this->target->has($id, $domain)) {
```

Comment:
@aitboudad there is a separate method to get obsolete messages (and this is broken after this PR)